### PR TITLE
Added exporting nodes reservation metrics for Autoscaling sink

### PR DESCRIPTION
**Pull request by piosz**
cc @vmarmol @vishh @jszczepkowski

### CODE REVIEW

**Comment by vmarmol** [comment]
ok to test

File: sinks/gcmautoscaling/driver.go [hunk 1]
```diff
@@ -52,11 +52,19 @@ type utilizationMetric struct {
 var autoscalingMetrics = map[string]utilizationMetric{
 	"cpu/usage": {
 		name:        "cpu/node_utilization",
-		description: "Cpu utilization as a percentage of node capacity",
+		description: "Cpu utilization as a share of node capacity",
```
Comment (vmarmol) [inline sinks/gcmautoscaling/driver.go]: What do you mean by shares? What units?

Comment (piosz) [inline sinks/gcmautoscaling/driver.go]: I mean total usage divided by node capacity. It's the number from range [0, 1] without any unit.

Comment (vmarmol) [inline sinks/gcmautoscaling/driver.go]: Just curious about the units :)

File: sinks/gcmautoscaling/driver.go [hunk 2]
```diff
@@ -88,29 +108,90 @@ func isNode(metric *sink_api.Point) bool {
...
+		LabelGCEResourceType.Key: "instance",
+	}
+}
+
+func (self *gcmAutocalingSink) getNewValue(metric *sink_api.Point, ts *gcm.Timeseries) *float64 {
+	host := metric.Labels[sink_api.LabelHostname.Key]
+
+	var val float64
+	switch metric.Name {
+	case "cpu/usage":
```
Comment (vmarmol) [inline sinks/gcmautoscaling/driver.go]: Seems like we should start using defined values for these strings. We can do in another PR if we'd like.

File: sinks/gcmautoscaling/driver.go [hunk 3]
```diff
@@ -88,29 +108,90 @@ func isNode(metric *sink_api.Point) bool {
...
+			if metric.Name == "cpu/limit" {
+				self.cpuReservation[hostTime{host, metric.End}] += value
+			} else if metric.Name == "memory/limit" {
+				self.memReservation[hostTime{host, metric.End}] += value
+			}
+		}
+	}
+}
+
+func getLabels(metric *sink_api.Point) map[string]string {
```
Comment (vmarmol) [inline sinks/gcmautoscaling/driver.go]: Can we document this function and the one below it?

**Comment by vmarmol** [comment]
LGTM, sorry for the delay. Just minor edits. Will merge to unblock you, please send another PR for the fixes :)

**Comment by piosz** [comment]
No problem. Thanks for review! Added follow up PR.
